apps/main/src/llamalend/features/bands-chart/chartOptions.ts

```diff
@@ -0,0 +1,242 @@
+import type { EChartsOption } from 'echarts-for-react'
+import { formatNumber } from '@ui/utils'
+import { generateMarkLines, createLabelStyle } from './markLines'
+import { ChartDataPoint, BandsChartPalette, DerivedChartData, UserBandsPriceRange } from './types'
+
+//
+// Custom series renderer to draw a rectangle spanning [p_down, p_up] with a given width and start offset.
+// Data value layout per item: [median, startX, widthX, p_down, p_up, isLiquidationNumeric]
+//
+const createCustomRectSeries = (
+  name: string,
+  color: string,
+  outlineColor: string,
+  data: Array<[number, number, number, number, number, number]>,
+  markArea?: Record<string, unknown> | null,
+  markLine?: Record<string, unknown> | null,
+) => ({
+  name,
+  type: 'custom' as const,
+  animation: false,
+  clip: false,
+  encode: { y: 0 },
+  data,
+  emphasis: {
+    focus: 'self',
+    itemStyle: {
+      opacity: 1,
+    },
+  },
+  renderItem: (_params: unknown, api: any) => {
+    const startX = api.value(1)
+    const widthX = api.value(2)
+    const pDown = api.value(3)
+    const pUp = api.value(4)
+    const isLiq = api.value(5) === 1
+
+    // Do not render when the logical width is zero or negative
+    if (!widthX || widthX <= 0) return null
+
+    const tl = api.coord([startX, pUp])
```

Comment:
omg what does `tl` `tr` `bl` and `pUp` mean? Please avoid abbreviations (even with stuff like `isLiq` it's not needed)